pkgs/applications/science/biology/bowtie2/default.nix

```diff
@@ -8,19 +8,37 @@ stdenv.mkDerivation rec {
     owner = "BenLangmead";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-HaiZmWU6akHXJVWBmCvkG2E61NDrAP7UIxx9DNCEZqE=";
+    hash = "sha256-aTRdisw9b0Zbo0XOgd9gxw/S2qmPq/QqMSY853nrz80=";
+    fetchSubmodules = true;
   };
 
   nativeBuildInputs = [ cmake ];
 
   buildInputs = [ tbb zlib python3 perl ];
 
+  cmakeFlags = lib.optional (!stdenv.hostPlatform.isx86) ["-DCMAKE_CXX_FLAGS=-I${src}/third_party"];
+
+  doInstallCheck = true;
+
+  installCheckPhase = ''
+    runHook preInstallCheck
+
+    $out/bin/bowtie2 -x $src/example/index/lambda_virus -U $src/example/reads/longreads.fq > /dev/null
+    $out/bin/bowtie2 -x $src/example/index/lambda_virus -1 $src/example/reads/reads_1.fq -2 $src/example/reads/reads_2.fq > /dev/null
+    $out/bin/bowtie2-build $src/example/reference/lambda_virus.fa lambda_virus > /dev/null
+    $out/bin/bowtie2-build --large-index $src/example/reference/lambda_virus.fa lamba_virus_large > /dev/null
+    $out/bin/bowtie2-inspect --summary $src/example/index/lambda_virus > /dev/null
+    $out/bin/bowtie2-inspect --large-index $src/example/index/lambda_virus > /dev/null
+
+    runHook postInstallCheck
+  '';
+
   meta = with lib; {
     description = "An ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences";
-    license = licenses.gpl3;
+    license = licenses.gpl3Plus;
     homepage = "http://bowtie-bio.sf.net/bowtie2";
+    changelog = "https://github.com/BenLangmead/bowtie2/releases/tag/v${version}";
     maintainers = with maintainers; [ rybern ];
     platforms = platforms.all;
-    broken = stdenv.isAarch64; # only x86 is supported
```

Comment:
doesn't build on aarch64-linux, maybe mark as broken on `stdenv.isAarch64 && stdenv.isLinux` ?

Reply:
Thanks for your review!
I don't remember much about this PR. I'll check it later.

Reply:
no rush :+1:

Reply:
We can now build on aarch64-linux.